crates/rpc/src/v05/method/get_transaction_receipt.rs

```diff
@@ -0,0 +1,780 @@
+        Declare(DeclareTransactionReceipt),
+        #[serde(rename = "L1_HANDLER")]
+        L1Handler(L1HandlerTransactionReceipt),
+        // FIXME regenesis: remove Deploy receipt type after regenesis
+        // We are keeping this type of receipt until regenesis
+        // only to support older pre-0.11.0 blocks
+        #[serde(rename = "DEPLOY")]
+        Deploy(DeployTransactionReceipt),
+        #[serde(rename = "DEPLOY_ACCOUNT")]
+        DeployAccount(DeployAccountTransactionReceipt),
+    }
+
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    #[cfg_attr(any(test, feature = "rpc-full-serde"), derive(serde::Deserialize))]
+    pub struct InvokeTransactionReceipt {
+        #[serde(flatten)]
+        pub common: CommonTransactionReceiptProperties,
+    }
+
+    #[serde_as]
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    #[cfg_attr(any(test, feature = "rpc-full-serde"), derive(serde::Deserialize))]
+    pub struct CommonTransactionReceiptProperties {
+        #[serde_as(as = "RpcFelt")]
+        pub transaction_hash: TransactionHash,
+        pub actual_fee: Fee,
+        #[serde_as(as = "RpcFelt")]
+        pub block_hash: BlockHash,
+        pub block_number: BlockNumber,
+        pub messages_sent: Vec<MessageToL1>,
+        pub events: Vec<Event>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub revert_reason: Option<String>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub execution_resources: Option<ExecutionResourcesProperties>,
+        pub execution_status: ExecutionStatus,
+        pub finality_status: FinalityStatus,
+    }
+
+    #[serde_as]
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    #[cfg_attr(any(test, feature = "rpc-full-serde"), derive(serde::Deserialize))]
+    pub struct CommonPendingTransactionReceiptProperties {
+        pub transaction_hash: TransactionHash,
+        pub actual_fee: Fee,
+        pub messages_sent: Vec<MessageToL1>,
+        pub events: Vec<Event>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub revert_reason: Option<String>,
+        pub execution_status: ExecutionStatus,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub execution_resources: Option<ExecutionResourcesProperties>,
+        pub finality_status: FinalityStatus,
+    }
+
+    /// Similar to `ExecutionResources`, with irrelevant properties stripped.
+    #[serde_as]
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    #[cfg_attr(any(test, feature = "rpc-full-serde"), derive(serde::Deserialize))]
+    pub struct ExecutionResourcesProperties {
+        pub steps: String,
+        pub memory_holes: String,
+        pub range_check_builtin: String,
+        pub pedersen_builtin: String,
+        pub poseidon_builtin: String,
+        pub ec_op_builtin: String,
+        pub ecdsa_builtin: String,
+        pub bitwise_builtin: String,
+        pub keccak_builtin: String,
```

Comment:
These are `NUM_AS_HEX` according to the specification. We have a somewhat in-elegent solution for this in our serde helper crate: `U64AsHexStr(u64)` which has the correct serde serialization.

i.e. should serialize to `"0x123"` and not `"123"`.

A PR to change these to integers has already been merged in the specification; but of course that doesn't apply to v0.5 😞